csrc/kernel_cache.cpp

```diff
@@ -196,17 +196,20 @@ class ArgumentManager {
       Val* output = group_outputs[group_out_i];
       const PolymorphicValue*& runtime_output = tensor_map_[output];
       if (runtime_output != nullptr) {
-        // A trivial forwarding output shares the same `Val*` as an input, so we
-        // simply map it to the same runtime output.
+        // A trivial forwarding output or a dupliated output shares the same
+        // `Val*` as another fusion input/output. In those cases, we keep
+        // mapping it to the same runtime output.
         continue;
       }
 
+      PolymorphicValue group_runtime_output;
       if constexpr (std::is_pointer_v<
                         decltype(group_runtime_outputs[group_out_i])>) {
-        fusion_args_.push(*group_runtime_outputs[group_out_i]);
+        group_runtime_output = *group_runtime_outputs[group_out_i];
       } else {
-        fusion_args_.push(group_runtime_outputs[group_out_i]);
+        group_runtime_output = group_runtime_outputs[group_out_i];
       }
+      fusion_args_.push(std::move(group_runtime_output));
```

Comment:
out of curiosity, is this pattern better than the previous push to fusion_args_ directly?

Reply:
I forgot to revert this. It was a side effect of a lateral change that I later reverted.